### PR TITLE
Closes #9530: Don't crash on failed avatar fetches

### LINKED ISSUE #9530: ConnectException: Failed to connect to firefoxusercontent.com/13.224.164.41:443
https://sentry.prod.mozaws.net/operations/fenix/issues/7591794/

```
ConnectException: Failed to connect to firefoxusercontent.com/13.224.164.41:443
    at org.mozilla.fenix.settings.SettingsFragment$updateAccountUIState$$inlined$let$lambda$1.invokeSuspend(SettingsFragment.kt:7)
...
(12 additional frame(s) were not displayed)

Failed to connect to firefoxusercontent.com/13.224.164.41:443
```

### PR DESCRIPTION
**Pull request by grigoryk**
### Pull Request checklist

- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### After merge
- [ ] **Milestone**: Make sure issues finished by this pull request are added to the [milestone](https://github.com/mozilla-mobile/fenix/milestones) of the version currently in development.

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

### CODE REVIEW

**Comment by grigoryk** [comment]
We're also not using the configured fetch client here, either... That's worth fixing as well.

File: app/src/main/java/org/mozilla/fenix/ext/String.kt [hunk 1]
```diff
@@ -113,17 +116,23 @@ fun String.simplifiedUrl(): String {
...
-    val bitmap = BitmapFactory.decodeStream(avatarUrl.openConnection().getInputStream())
-    val roundedBitmapDrawable = RoundedBitmapDrawableFactory.create(context.resources, bitmap)
-    roundedBitmapDrawable.isCircular = true
-    roundedBitmapDrawable.setAntiAlias(true)
-    return roundedBitmapDrawable
+}
+
+suspend fun bitmapForUrl(url: String, client: Client): Bitmap? = withContext(Dispatchers.IO) {
+    // TODO cache this image, see https://github.com/mozilla-mobile/fenix/issues/9531
+    // Code below will cache it in Gecko's cache, which ensures that as long as we've fetched it once,
```
Comment (ekager) [inline app/src/main/java/org/mozilla/fenix/ext/String.kt]: When a user changes their account photo desktop, will we get a new URL? aka will this update or just keep using the cache?

Comment (grigoryk) [reply app/src/main/java/org/mozilla/fenix/ext/String.kt]: Not sure, I'll test this.

Comment (grigoryk) [reply app/src/main/java/org/mozilla/fenix/ext/String.kt]: Works correctly after app restart, when we refetch the profile.

After a profile change we have a push message coming in, telling the app that the profile changed; once we support it, we can refetch the profile without having to wait for a restart.

**Comment by grigoryk** [comment]
Another problem I just noticed - we're going through the avatar code twice. Not so bad now that it's using a cache (before, we'll kick off two network requests), but we're still doing bunch of unnecessary work.

**Comment by grigoryk** [comment]
That may be a larger refactor, so let's just land this for now.

**Pull request merged**